### PR TITLE
feat: add ckbunker install script

### PR DESCRIPTION
**Pull request by openoms**
Related: https://github.com/rootzoll/raspiblitz/issues/1062

Initial script, needs testing and see the first setup flow.

### CODE REVIEW

File: home.admin/_commands.sh [hunk 1]
```diff
@@ -311,6 +312,19 @@ function jm() {
   fi
 }
 
+# command: ckbunker
+# switch to the joinmarket user for the JoininBox menu
```
Comment (vv01f) [inline home.admin/_commands.sh]: is that some outdated leftover?

File: home.admin/config.scripts/bonus.ckbunker.sh [hunk 2]
```diff
@@ -0,0 +1,288 @@
+#!/bin/bash
+
+# https://github.com/lightninglabs/ckbunker/releases-
+pinnedVersion="bf08623875b576c4bc4498dc68e749cdf6b5de31"
+
+# command info
+if [ $# -eq 0 ] || [ "$1" = "-h" ] || [ "$1" = "-help" ]; then
+  echo "# Config script to switch the CKbunker on or off"
+  echo "# Installs CKBunker https://github.com/Coldcard/ckbunker/commit/$pinnedVersion"
+  echo "# bonus.${APPID}.sh status   -> status information (key=value)"
```
Comment (vv01f) [inline home.admin/config.scripts/bonus.ckbunker.sh]: variable used before declaration

File: home.admin/_commands.sh (hunk 1 above)
Comment (openoms) [reply home.admin/_commands.sh]: cleaned with [66bf3e3](https://github.com/rootzoll/raspiblitz/pull/3148/commits/66bf3e31735aa32720e8675f32a87be77ff4b159)

File: home.admin/config.scripts/bonus.ckbunker.sh (hunk 2 above)
Comment (openoms) [reply home.admin/config.scripts/bonus.ckbunker.sh]: cleaned with [66bf3e3](https://github.com/rootzoll/raspiblitz/pull/3148/commits/66bf3e31735aa32720e8675f32a87be77ff4b159)

**Comment by rootzoll** [comment]
@openoms for which release are you targeting this?

**Comment by openoms** [comment]
whenever it's ready. If people want it might get some help testing too.
I have the coldcards so can do the basics even if it is just CLI only for v1.8.0.

**Comment by rootzoll** [comment]
So I mark it for v1.8.0 but we will see close before final release if we include CLI or not.

**Comment by openoms** [comment]
The install script is functional now, but getting an error running 
`ckbunker run`
or 
`ckbunker setup`

```
(ENV) ckbunker@z420:~$ ckbunker setup
Traceback (most recent call last):
  File "/home/ckbunker/.local/bin/ckbunker", line 33, in <module>
    sys.exit(load_entry_point('bunker', 'console_scripts', 'ckbunker')())
  File "/usr/lib/python3/dist-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3/dist-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/ckbunker/ckbunker/main.py", line 76, in setup_hsm
    asyncio.run(startup(True, local, config_file, None), debug=True)
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/ckbunker/ckbunker/main.py", line 109, in startup
    import webapp
  File "/home/ckbunker/ckbunker/webapp.py", line 5, in <module>
    import sys, os, asyncio, logging, aiohttp_jinja2, jinja2, time, weakref, re
  File "/home/ckbunker/.local/lib/python3.9/site-packages/aiohttp_jinja2/__init__.py", line 29, in <module>
    from .helpers import GLOBAL_HELPERS
  File "/home/ckbunker/.local/lib/python3.9/site-packages/aiohttp_jinja2/helpers.py", line 23, in <module>
    @jinja2.pass_context
AttributeError: module 'jinja2' has no attribute 'pass_context'
sys:1: RuntimeWarning: coroutine 'Connection.run' was never awaited
sys:1: RuntimeWarning: coroutine 'TorViaStem.startup' was never awaited
```

**Comment by openoms** [comment]
The install script is functional now., but the first setup is still too complicated.
Looking for a way to check if the ColdCard has been set up before and can use `ckbunker run` (as in the systemd service)
Otherwise:
```
# To set up on first use:
# stop the ckbunker service:
sudo systemctl stop ckbunker
# switch to the user
sudo su - ckbunker
# run:
ckbunker setup
# open in your local web browser:
https://${localIP}:${PORT_SSL}/setup
# and follow the guide at:
https://ckbunker.com/setup.html
```

**Comment by openoms** [comment]
First tested by connecting an already initiliazed CC which was used with CKbunker before:
```
(ENV) ckbunker@raspberrypi:~/ckbunker $ ckbunker run                                                               
[04/06/2022-20:09:34] Web server at:    http://localhost:9823                                                      
[04/06/2022-20:09:35] Connecting to Coldcard.                                                                      
[04/06/2022-20:09:35] Tord version: 0.4.7.7                                                                        
[04/06/2022-20:09:35] Found Coldcard xx. USB encryption version: 1                                       
[04/06/2022-20:09:36] Connected to Coldcard xx.                                                          
[04/06/2022-20:09:37] Coldcard now in HSM mode. Fetching storage locker.                                           
[04/06/2022-20:09:40] ./data/bp-xx.dat: not found (probably fine)                                    
[04/06/2022-20:09:40] Unable to read bunker settings for this Coldcard; forging on                                 
[04/06/2022-20:09:40] Captured CC's summary of the policy                                                          
[04/06/2022-20:09:40] Saved bunker settings to: ./data/bp-xx.dat                                     
[04/06/2022-20:10:04] ping ok    
```

```
(ENV) ckbunker@raspberrypi:~/ckbunker $ ls -al data                                   
total 20                                                                              
drwxr-xr-x  2 ckbunker ckbunker 4096 Jun  4 20:09 .                                   
drwxr-xr-x 10 ckbunker ckbunker 4096 Jun  4 08:38 ..                                  
-rw-r--r--  1 ckbunker ckbunker  507 Jun  4 20:09 bp-xx.dat             
-rw-r--r--  1 ckbunker ckbunker    6 Jun  4 08:25 .gitignore                          
-rw-r--r--  1 ckbunker ckbunker  384 Jun  4 08:25 README.md
```

```
(ENV) ckbunker@raspberrypi:~/ckbunker $ cat data/README.md 
        

# Data Files

- this directory will hold data files for the Bunker
- they are encrypted with a private key held in the "storage locker" of a Coldcard
- contents include Tor hidden service private key and settings for Bunker
- filename based on key
- you may see unused junk accumulate in this directory; those are random keys that
  never got saved as a policy file for any Coldcard
```

**Comment by openoms** [comment]
Possible settings:

    # web server port                                                                                                                                                                                                                                                                                                         
    PORT_NUMBER = 9823                                                                                                                                                                                                                                                                                                        
                                                                                                                                                                                                                                                                                                                              
    # session idle time, before we kick you out and require re-auth (seconds)                                                                                                                                                                                                                                                 
    MAX_IDLE_TIME = 10*60                                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                                                              
    # max time between showing login page, and the would-be user entering something useful (seconds)                                                                                                                                                                                                                          
    MAX_LOGIN_WAIT_TIME = 5*60                                                                                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                                                                              
    # bogus fixed password to get started                                                                                                                                                                                                                                                                                     
    MASTER_PW = 'test1234'                                                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                              
    # default is harder captcha                                                                                                                                                                                                                                                                                               
    EASY_CAPTCHA = False                                                                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                                                              
    # default for "allow reboot of bunker"                                                                                                                                                                                                                                                                                    
    # - can you restart the bunker w/o restarting the Coldcard HSM?                                                                                                                                                                                                                                                           
    ALLOW_REBOOTS = True                                                                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                                                              
    # path to data files                                                                                                                                                                                                                                                                                                      
    DATA_FILES = './data'                                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                                                              
    # endpoint to use for sending txn; we assume it's Explora protocol (Blockstream.info)                                                                                                                                                                                                                                     
    EXPLORA = 'http://explorerzydxu5ecjrkwceayqybizmpjjznk5izmitf2modhcusuqlid.onion'                                                                                                                                                                                                                                         
                                                                                                                                                                                                                                                                                                                              
    # port number for local instance of tord                                                                                                                                                                                                                                                                                  
    # - will try 9051 and 9151                                                                                                                                                                                                                                                                                                
    # - but first /var/run/tor/control as unix socket                                                                                                                                                                                                                                                                         
    TORD_PORT = 'default'                                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                                                              
    # for broadcasting, socks proxy via Tord                                                                                                                                                                                                                                                                                  
    TOR_SOCKS = 'socks5h://127.0.0.1:9150'                                                                                                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                                                              
    # unix pipe for local Coldcard Simulator                                                                                                                                                                                                                                                                                  
    SIMULATOR_SOCK = '/tmp/ckcc-simulator.sock'                                                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                                              
    # delay between retries connecting to missing/awol Coldcard                                                                                                                                                                                                                                                               
    RECONNECT_DELAY = 10        # seconds between retries                                                                                                                                                                                                                                                                     
    PING_RATE = 15              # seconds between pings (CC status checks)                                                                                                                                                                                                                                                    
    USB_NCRY_VERSION = 0x01     # default ncry version is 1

**Comment by openoms** [comment]
```
$ config.scripts/bonus.ckbunker.sh menu
```
image

**Comment by rootzoll** [comment]
@openoms is this PR merge ready for v1.8.0?

**Comment by openoms** [comment]
Can work for CLI only, the installation was tested ok. Did not have time to improve the UX more, but it can be already helpful for someone looking to use ckbunker then we can have some suggestions coming how to make the initialization easier.

**Pull request merged**